qcodes/dataset/sqlite_storage_interface.py

```diff
@@ -0,0 +1,30 @@
+from typing import Union, Dict, Sequence
+from numbers import Number
+from numpy import ndarray
+
+from .data_storage_interface import (
+    DataStorageInterface, VALUES)
+from .sqlite_base import (
+    connect, select_one_where, insert_values, insert_many_values)
+from qcodes.dataset.database import get_DB_location
+
+
+class SqliteStorageInterface(DataStorageInterface):
+    """
+    """
+    def __init__(self, guid: str, ds):
```

Comment:
The signature is not consistent with the ABC, should the ds be passed in the ABC too?
?

Reply:
I thought the storage interface instance should be an attribute of the dataset?

Reply:
This cyclic reference makes me uneasy. Can we not circumvent it?

Reply:
shall we fix the wrapping issue instead?

Reply:
From my understanding of abstract base classes there is no need for the constructor of a child class to have the same call signature as the base, on the contrary it is most likely to be different. But I am happy to learn from your experience @jenshnielsen, it would be great if we could have a brief chat about this.

@WilliamHPNielsen, @astafan8, I agree that this is not a good solution here with the dataset being passed to the init. As you mentioned this is a quick and dirty fix for the issue I wrote about on Slack. My intention was to provide an interface that passes the basic tests and serves as a basis for the development of the other back-ends. As we have tight schedule we should first get our prototypes up and running before we fix the wrapping issue.

Reply:
It not strictly disallowed but considered bad style and violates the [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle)